modules/mcuboot/CMakeLists.txt

```diff
@@ -6,14 +6,33 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
 
   include(${ZEPHYR_BASE}/../nrf/cmake/fw_zip.cmake)
 
-  function(sign to_sign_hex output_prefix offset sign_depends signed_hex_out)
-    set(op ${output_prefix})
-    set(signed_hex ${op}_signed.hex)
-    set(${signed_hex_out} ${signed_hex} PARENT_SCOPE)
-    set(to_sign_bin ${op}_to_sign.bin)
-    set(update_bin ${op}_update.bin)
-    set(moved_test_update_hex ${op}_moved_test_update.hex)
-    set(test_update_hex ${op}_test_update.hex)
+  function(sign)
+    # Signs a hex image
+    #
+    # Required arguments are:
+    # HEX_TO_SIGN - Hex image to sign
+    # SIGNED_PREFIX - Prefix of the output image
+    # OFFSET - offset
+    # HEX_SIGNED - Signed hex output image
+    # DEPENDS - One or more dependencies for signing the hex image
```

Comment:
Thanks, always good to document functions.

Could you maybe also describe with options are mandatory and which are optional.